Installation Guide

```diff
@@ -0,0 +1,266 @@
+
+### **Section 1 -  System Components Installation**
+
+#### Step 1: Install the IUS repository
+The IUS repository provides newer versions of some software in the official CentOS and Red Hat repositories. The IUS repository depends on the EPEL repository.
+```
+# yum -y install https://centos7.iuscommunity.org/ius-release.rpm
+```
+
+#### Step 2: Installation of  Git, Python 3.6 and All Requirements
+```
+# yum install python36u python36u-devel python36u-setuptools git gcc libxml2-devel libxslt-devel libffi-devel graphviz openssl-devel redhat-rpm-config -y
+```
+
+#### **Step 3: Install PIP**
+```
+easy_install-3.6 pip
+```
+
+#### **Step 4: Create Python Symlink**
+```
+# sudo ln -s /usr/bin/python3.6 /usr/bin/python3
+```
+
+#### **Step 4: Check Python 3 Version**
+```
+# python3 --version
+```
+
+### **Section 2 -  MySQL Database Configurations**
+
+#### Step 1: Add MariaDB Yum Repository
+
+Create the MariaDB repo file by issuing the following command
+```
+# sudo vi /etc/yum.repos.d/MariaDB.repo
+```
+
+Paste the following contents in MariaDB.repo and save the file
+```
+[mariadb]
+name = MariaDB
+baseurl = http://yum.mariadb.org/10.1/centos7-amd64
+gpgkey=https://yum.mariadb.org/RPM-GPG-KEY-MariaDB
+gpgcheck=1
+```
+
+#### Step 2 - Install MariaDB Server, Client and Development Tools
+
+```
+# sudo yum install MariaDB-server MariaDB-devel MariaDB-client -y
+```
+Once the installation is complete, we'll start the MariaDB daemon with the following command:
+
+```
+# systemctl start mariadb
+```
+To ensure that the MariaDB daemon is started, issue the following command
+```
+# sudo systemctl status mariadb
+```
+
+Finally after ensuring that MariaDB is able to start successfully, we will use the systemctl enable command to ensure that MariaDB will start at boot.
+```
+# sudo systemctl enable mariadb
+```
+
+#### Step 3 - Securing the MariaDB Server
+MariaDB includes a security script to modify some of the less secure default options for items such as remote root logins and sample users. Issue the following command to execute the security script:
+```
+# sudo mysql_secure_installation
+```
+The script provides a detailed explanation for every step. The first prompts asks for the root password, which hasn't been set so we'll press ENTER as it recommends. Next, we'll be prompted to set that root password, which we'll do.
+
+Then, we'll accept all the security suggestions by pressing Y and then ENTER for the remaining prompts, which will remove anonymous users, disallow remote root login, remove the test database, and reload the privilege tables.
+
+#### Step 4 - Create the promgen Database
+Login as root user with the password set earlier at step 2
+```
+# mysql -u root -p
+```
+
+Create promgen database
+```
+> CREATE DATABASE promgen;
+```
+
+Exit MySQL command utility
+```
+> quit
+```
+
+### Section 3 -  Deploying Promgen Application
+
+#### Step 1: Install the Required Python Packages
+```
+# pip3 install mysqlclient
```

Comment:
Typically I've always recommended using a virtualenv instead of directly installing into the system Python. Is there a reason for installing it to the system python?

Reply:
I am not very familiar with virtualenv, will retest with virtualenv and feedback